drivers/clock_control/clock_control_bl60x.c

```diff
@@ -0,0 +1,923 @@
+	} else if (tmp == 0) {
+		return 48 * 1000 * 1000 / (hclk_div + 1);
+	}
+	return -EINVAL;
+}
+
+/* most peripherals clock */
+static uint32_t clock_control_bl60x_get_bclk(const struct device *dev)
+{
+	uint32_t tmp;
+	uint32_t clock_id;
+
+	tmp = sys_read32(GLB_BASE + GLB_CLK_CFG0_OFFSET);
+	tmp = (tmp & GLB_REG_BCLK_DIV_MSK) >> GLB_REG_BCLK_DIV_POS;
+	clock_id = clock_control_bl60x_get_clk(dev);
+	return clock_id / (tmp + 1);
+}
+
+static uint32_t clock_control_bl60x_mtimer_get_clk_src_div(const struct device *dev)
+{
+	return clock_control_bl60x_get_bclk(dev) / 1000 / 1000 - 1;
+}
+
+static void clock_control_bl60x_cache_2T(bool yes)
+{
+	uint32_t tmp;
+
+	tmp = sys_read32(L1C_BASE + L1C_CONFIG_OFFSET);
+
+	if (yes) {
+		tmp |= L1C_IROM_2T_ACCESS_MSK;
+	} else {
+		tmp &= ~L1C_IROM_2T_ACCESS_MSK;
+	}
+
+	sys_write32(tmp, L1C_BASE + L1C_CONFIG_OFFSET);
+}
+
+/* HCLK: 0
+ * PLL120M: 1
+ */
+static void clock_control_bl60x_set_PKA_clock(uint32_t pka_clock)
+{
+	uint32_t tmp;
+
+	tmp = sys_read32(GLB_BASE + GLB_SWRST_CFG2_OFFSET);
+	tmp = (tmp & GLB_PKA_CLK_SEL_UMSK) | (pka_clock << GLB_PKA_CLK_SEL_POS);
+	sys_write32(tmp, GLB_BASE + GLB_SWRST_CFG2_OFFSET);
+}
+
+static void clock_control_bl60x_init_root_as_pll(const struct device *dev)
+{
+	struct clock_control_bl60x_data *data = dev->data;
+	const struct clock_control_bl60x_config *config = dev->config;
+	uint32_t tmp;
+
+	clock_control_bl60x_init_pll(data->pll.source, config->crystal_frequency);
+
+	/* enable all 'PDS' clocks */
+	tmp = sys_read32(PDS_BASE + PDS_CLKPLL_OUTPUT_EN_OFFSET);
+	tmp |= 0x1FF;
+	sys_write32(tmp, PDS_BASE + PDS_CLKPLL_OUTPUT_EN_OFFSET);
+
+	/* glb enable pll actual? */
+	tmp = sys_read32(GLB_BASE + GLB_CLK_CFG0_OFFSET);
+	tmp = (tmp & GLB_REG_PLL_EN_UMSK) | ((uint32_t)(1) << GLB_REG_PLL_EN_POS);
+	sys_write32(tmp, GLB_BASE + GLB_CLK_CFG0_OFFSET);
+
+	clock_control_bl60x_select_PLL(data->root.pll_select);
+
+	if (data->pll.source == bl60x_clkid_clk_crystal) {
+		clock_control_bl60x_set_root_clock(3);
+	} else {
+		clock_control_bl60x_set_root_clock(2);
+	}
+
+	if (clock_control_bl60x_get_clk(dev) > 120000000) {
+		clock_control_bl60x_cache_2T(true);
+	}
+
+	sys_write32(clock_control_bl60x_get_clk(dev), CORECLOCKREGISTER);
+	clock_control_bl60x_set_PKA_clock(1);
+}
+
+static void clock_control_bl60x_init_root_as_crystal(const struct device *dev)
+{
+	clock_control_bl60x_set_root_clock(1);
+	sys_write32(clock_control_bl60x_get_clk(dev), CORECLOCKREGISTER);
+}
+
+static int clock_control_bl60x_update_root(const struct device *dev)
+{
+	struct clock_control_bl60x_data *data = dev->data;
+	uint32_t tmp;
+	int ret;
+
+	/* make sure all clocks are enabled */
+	tmp = sys_read32(GLB_BASE + GLB_CLK_CFG0_OFFSET);
+	tmp = (tmp & GLB_REG_BCLK_EN_UMSK) | ((uint32_t)(1) << GLB_REG_BCLK_EN_POS);
+	tmp = (tmp & GLB_REG_HCLK_EN_UMSK) | ((uint32_t)(1) << GLB_REG_HCLK_EN_POS);
+	tmp = (tmp & GLB_REG_FCLK_EN_UMSK) | ((uint32_t)(1) << GLB_REG_FCLK_EN_POS);
```

Comment:
This should have the same effect, but looks cleaner
```suggestion
	tmp = (tmp & GLB_REG_BCLK_EN_UMSK) | (1U << GLB_REG_BCLK_EN_POS);
	tmp = (tmp & GLB_REG_HCLK_EN_UMSK) | (1U << GLB_REG_HCLK_EN_POS);
	tmp = (tmp & GLB_REG_FCLK_EN_UMSK) | (1U << GLB_REG_FCLK_EN_POS);
```

Reply:
Done